probe-rs-target/src/chip.rs

```diff
@@ -1,6 +1,42 @@
 use super::memory::MemoryRegion;
 use crate::{serialize::hex_option, CoreType};
 use serde::{Deserialize, Serialize};
+
+/// Represents a DAP scan chain element.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDap {
+    /// Unique name of the DAP
+    pub name: Option<String>,
+    /// Specifies the type of the DAP (e.g. ARMCS-DP)
+    pub dap_type: Option<String>,
+    /// Specifies the IR length of the DAP (default value: 4).
+    pub ir_len: Option<u8>,
+    /// List of protocols via which the DAP can be use. Possible values are SWD, JTAG, and cJTAG.
+    pub protocols: Option<Vec<String>>,
+    /// Specifies the SW-DPv2 TARGETSEL register value that selects this DAP in a serial-wire debug multi-drop system
+    pub targetsel: Option<u8>,
+}
+
+/// Represents a non DAP device in the scan chain. This can be other TAPs.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDevice {
+    /// Unique name of the device
+    pub name: Option<String>,
+    /// Specifies the type of the device (e.g. ARMCS-DP)
+    pub device_type: Option<String>,
```

Comment:
Same comment as above :) What's the difference between a device and a DAP? Is the device the unification of multiple DAPs? then maybe we should have the structure as such? Again, I am not sure here, just raising some question to make your life harder ;P